src/use_anchor_sensitive_click_handler/index.es

```diff
@@ -0,0 +1,26 @@
+import { useHistory } from 'react-router';
+import { stripOriginFromUrl } from 'nebenan-helpers/lib/routes';
+import { useCallback } from 'react';
+
+const useAnchorSensitiveClickHandler = (callback) => {
+  const history = useHistory();
+
+  return useCallback((event) => {
+    const { target } = event;
+    const href = target.getAttribute('href');
+    const isLink = target.tagName === 'A';
+
+    if (isLink && href) {
+      event.preventDefault();
+      const isSameDomain = href.startsWith(global.location.origin);
+      const path = stripOriginFromUrl(href, global.location.origin);
+      if (isSameDomain) history.push(path);
+      else global.open(href);
```

Comment:
We should parse the anchor some more to respect stuff like `noopener`

Reply:
-> `useSmartClickHandler`?

we run that on generated content, like markdown or smartcontent, we don't need to care about attributes